### PR TITLE
Bump python dependencies

### PR DESCRIPTION
**Pull request by hanefi**
This PR updates the dependencies to upgrade and organize all the dependencies.

dependabot PRs:
Closes #96 #97 #98 #99 #100 #101 #105 #107 #108

alternative solution:
Closes #120

### CODE REVIEW

File: packaging_automation/.python-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.8.5
```
Comment (gurkanindibay) [inline packaging_automation/.python-version:1]: I think you used pyenv to fix the python version. However we are not using pyenv in our CI/CD exec environment. If we use, we need to add a pre execution step to install it which will increase the time needed to finish he job.
I think adding pyenv is not necessary right now

File: packaging_automation/requirements.in [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
+psycopg2
 pycurl
 PyGithub
 pytest
-python-string-utils
-requests
-urllib3
 python-gnupg
-parameters_validation
+python-string-utils
```
Comment (gurkanindibay) [inline packaging_automation/requirements.in:11]: I see that order has changed. has hhanging the order affected the versions of libraries in requrements.txt?

File: packaging_automation/.python-version (hunk 1 above)
Comment (hanefi) [reply packaging_automation/.python-version:1]: I think we do not need to change anything on the CI. It's good practice to store the supported python versions in a python project. The file can stay there for those that wish to setup a virtual environment.

I may as well remove it. Your call.

File: packaging_automation/requirements.in (hunk 2 above)
Comment (hanefi) [reply packaging_automation/requirements.in:11]: No.

If you check the order of my commits, you will see that:
1. I ran `pip-compile --upgrade` to bump package versions,
2. Sort the dependencies in `requirements.in` manually, and remove extra blank lines and spaces.
3. Create `.python-version` to store the supported python version alongside the code.

When I run `pip-compile --upgrade` again after sorting the entries in `requirements.in`, `requirements.txt` file remains unchanged.

File: packaging_automation/.python-version (hunk 1 above)
Comment (gurkanindibay) [reply packaging_automation/.python-version:1]: I got it. No need to remove that file.

**Review by gurkanindibay — APPROVED** [review]

**Pull request merged**